metadata-ingestion/src/datahub/ingestion/source/sql/two_tier_sql_source.py

```diff
@@ -123,11 +124,18 @@ def get_inspectors(self):
             else:
                 databases = inspector.get_schema_names()
             for db in databases:
-                if self.config.database_pattern.allowed(db):
-                    url = self.config.get_sql_alchemy_url(current_db=db)
-                    with create_engine(url, **self.config.options).connect() as conn:
-                        inspector = inspect(conn)
-                        yield inspector
+                try:
+                    if self.config.database_pattern.allowed(db):
+                        url = self.config.get_sql_alchemy_url(current_db=db)
+                        with create_engine(
+                            url, **self.config.options
+                        ).connect() as conn:
+                            inspector = inspect(conn)
+                            yield inspector
+                except Exception as e:
+                    logger.debug(
+                        f"Failed to create connection with Database {db} -> {''.join(traceback.format_tb(e.__traceback__))}"
+                    )
```

Comment:
this should be a `self.error`

also can use `logger.debug(..., exc_info=e)` instead of doing that traceback thing

Reply:
I have added `logger.exception(...)` as I want to get the full stack trace because just the exception message is not helpful in fixing the errors